bin/e2e-test-client/src/test_context.rs

```diff
@@ -0,0 +1,191 @@
+        Output,
+        TransactionBuilder,
+        TxId,
+        UniqueIdentifier,
+        UtxoId,
+    },
+    fuel_types::{
+        Address,
+        AssetId,
+    },
+    fuel_vm::SecretKey,
+};
+
+use crate::config::{
+    ClientConfig,
+    SuiteConfig,
+};
+
+// The base amount needed to cover the cost of a simple transaction
+pub const BASE_AMOUNT: u64 = 10_000;
+
+pub struct TestContext {
+    pub alice: Wallet,
+    pub bob: Wallet,
+    pub config: SuiteConfig,
+}
+
+impl TestContext {
+    pub fn new(config: SuiteConfig) -> Self {
+        let alice_client = Self::new_client(config.endpoint.clone(), &config.wallet_a);
+        let bob_client = Self::new_client(config.endpoint.clone(), &config.wallet_b);
+        Self {
+            alice: Wallet::new(config.wallet_a.secret, alice_client),
+            bob: Wallet::new(config.wallet_b.secret, bob_client),
+            config,
+        }
+    }
+
+    fn new_client(default_endpoint: String, wallet: &ClientConfig) -> FuelClient {
+        FuelClient::new(wallet.endpoint.clone().unwrap_or(default_endpoint)).unwrap()
+    }
+}
+
+pub struct Wallet {
+    pub secret: SecretKey,
+    pub address: Address,
+    pub client: FuelClient,
+}
+
+impl Wallet {
+    pub fn new(secret: SecretKey, client: FuelClient) -> Self {
+        let public_key: PublicKey = (&secret).into();
+        let address = Input::owner(&public_key);
+        Self {
+            secret,
+            address,
+            client,
+        }
+    }
+
+    /// returns the balance associated with a wallet
+    pub async fn balance(&self, asset_id: Option<AssetId>) -> anyhow::Result<u64> {
+        self.client
+            .balance(
+                &self.address.to_string(),
+                Some(asset_id.unwrap_or_default().to_string().as_str()),
+            )
+            .await
+            .context("failed to retrieve balance")
+    }
+
+    /// Checks if wallet has a coin (regardless of spent status)
+    pub async fn owns_coin(&self, utxo_id: UtxoId) -> anyhow::Result<bool> {
+        let mut first_page = true;
+        let mut results = vec![];
+
+        while first_page || !results.is_empty() {
+            first_page = false;
+            results = self
+                .client
+                .coins(
+                    &self.address.to_string(),
+                    None,
+                    PaginationRequest {
+                        cursor: None,
+                        results: 100,
+                        direction: PageDirection::Forward,
+                    },
+                )
+                .await?
+                .results;
+            // check if page has the utxos we're looking for
+            if results
+                .iter()
+                .any(|coin| UtxoId::from(coin.utxo_id.clone()) == utxo_id)
+            {
+                return Ok(true)
+            }
+        }
+
+        Ok(false)
+    }
+
+    /// Transfers coins from this wallet to another
+    pub async fn transfer(
```

Comment:
Hmm, I hope we will not bring the logic of the SDK into our e2e tests in the future=)